src/components/shared/ARView/index.js

```diff
@@ -1,33 +1,57 @@
+// @flow
+
 import React, { Component } from "react";
-import { Text, View } from "react-native";
+import { View, Text } from "react-native";
 import { ViroARSceneNavigator, ViroUtils } from "react-viro";
+import CameraService from "../../../services/cameraService";
+import LangService from "../../../services/langService";
 
 import MarkerScene from "./MarkerScene";
 
 const { isARSupportedOnDevice } = ViroUtils;
 
-export default class ARView extends Component {
-  constructor(props) {
+const ARState = {
+  CAMERA_DISABLED: "AR_CAMERA_DISABLED",
+  CHECKING: "AR_CHECKING",
+  TRANSIENT: "AR_TRANSIENT",
+  SUPPORTED: "AR_SUPPORTED",
+  UNKNOWN: "AR_UNKNOWN",
+  UNSUPPORTED: "AR_UNSUPPORTED",
+};
+
+type Props = {}
+
+type State = {
+  arSupported: boolean,
+  arState: string
+}
+
+export default class ARView extends Component<Props, State> {
+  constructor(props: Props) {
     super(props);
 
-    this.state = {
-      arSupported: true,
-      reason: "",
-    };
+    this.state = { arSupported: false, arState: ARState.CHECKING };
 
-    const unsupportedCallback = reason => this.setState({ arSupported: false, reason });
-    const supportedCallback = () => this.setState({ arSupported: true });
+    CameraService
```

Comment:
Even though this is an async call I don't think it's good to set the state from the constructor. Think it would make more sense to place the check in `componentDidMount`